public/api/conf/1.0/application.raml

```diff
@@ -93,103 +93,103 @@ documentation:
+        description: Retrieve all Breathing Space periods for the given NINO.
+        (annotations.scope): "read:breathing-space-periods"
+        securedBy: [ sec.oauth_2_0: { scopes: [ "read:breathing-space-periods" ] } ]
+        headers:
+          Content-Type:
+            description: Specifies the content type of the request payload. Must be same as the example given.
+            type: string
+            required: true
+            example: application/json
+          Correlation-Id:
+            description: A unique UUID for this request
+            type: string
+            required: true
+            example: 24a8a24f-7126-4aa7-b690-259b2eaccaee
+          Request-Type:
+            description: |
+              Context flag for this request. Signifies if this request is the result of a human attended operation or an automated operation.
+
+              The two possible values are ‘DA2_BS_ATTENDED’ or ‘DA2_BS_UNATTENDED’.
+            type: string
+            required: true
+            example: DA2_BS_ATTENDED
+          Staff-Id:
+            description: |
+              The Staff identifier associated with the human attending the operation that generated this request.
+
+              Must be a 7 digit numeric value
+            type: string
+            required: true
+            example: "1234567"
+        responses:
+          200:
+            headers:
+              Correlation-Id:
+                description: A unique UUID for this request
+                type: string
+                required: true
+                example: 24a8a24f-7126-4aa7-b690-259b2eaccaee
+            body:
+              application/json:
+                type: !include schemas/CommonBsPeriodsResponse.json
+                examples:
+                  examples-1:
+                    description: Multiple periods returned
+                    value: !include examples/GetBsPeriodsResponse.json
+
+          400:
+            body:
+              application/json:
+                type: types.errorResponse
+                examples:
+                  missingHeader:
+                    description: One or more of the required headers are missing
+                    value:
+                      code: MISSING_HEADER
+                  invalidHeader:
+                    description: One or more of the header values are invalid
+                    value:
+                      code: INVALID_HEADER
+                  invalidNino:
+                    description: Invalid NINO format
+                    value:
+                      code: INVALID_NINO
+
+          404:
+            body:
+              application/json:
+                type: types.errorResponse
+                examples:
+                  resourceNotFound:
+                    description: The supplied NINO was not found
+                    value:
+                      code: RESOURCE_NOT_FOUND
+
+          502:
```

Comment:
Shouldn't 502 not added here but in a "general errors" section ?

Reply:
502 is not covered in the API Platform general errors reference page: https://developer.qa.tax.service.gov.uk/api-documentation/docs/reference-guide#errors

We should never return this ourselves, if we get this from IF we should return 500. Have a look at some other api project  implementations to see what they do.

Reply:
Ok. I'll remove it then from `handleUpstreamError` and as `Error` enum.